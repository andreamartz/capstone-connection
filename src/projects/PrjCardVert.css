.tag {
	margin-right: 0.65rem;
}
.HTML {
	color: #9c27b0;
}
.CSS {
	color: #ff4025;
}
.JS {
	color: #ff9c2a;
}
.API {
	color: #006400;
}
