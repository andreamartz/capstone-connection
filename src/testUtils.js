import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  id: 1,
  username: "testuser",
  firstName: "testfirst",
  lastName: "testlast",
  bio: "This is my bio",
  email: "test@test.net",
  photoUrl: null,
};

const UserProvider =
    ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
