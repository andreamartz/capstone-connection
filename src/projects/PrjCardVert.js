import React from "react";
import clsx from 'clsx';
import Box from '@material-ui/core/Box';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import CardActions from '@material-ui/core/CardActions';
import Divider from '@material-ui/core/Divider';
import { Link as MuiLink } from '@material-ui/core';
import { Link as ReactRouterDomLink } from "react-router-dom";
import { makeStyles } from '@material-ui/core/styles';
import { Avatar, IconButton, Typography } from '@material-ui/core';
import FavoriteBorderIcon from '@material-ui/icons/FavoriteBorder';
import ModeCommentOutlinedIcon from '@material-ui/icons/ModeCommentOutlined';
import Tags from '../tags/Tags';
import { pgTimeToDate } from '../utils';
import "./PrjCardVert.css";
import "../tags/Tags.css";


const useStyles = makeStyles((theme) => ({
  actions: {
    display: 'flex',
    justifyContent: 'space-between',
    paddingLeft: theme.spacing(2),
    paddingRight: theme.spacing(2),
    marginBottom: theme.spacing(1)
  }, 
  iconButton: {
    padding: '0'
  },
  date: {
    fontSize: theme.spacing(1.5)
  },
  avatar: {
    marginRight: theme.spacing(2)
  },
  large: {
    width: theme.spacing(6),
    height: theme.spacing(6),
  },
  creatorName: {
    textDecoration: 'none',
    '&:hover': {
      textDecoration: "underline",
    },
    fontWeight: 'bold'
  }
}));

function PrjCardVert({ 
  toggleLikeProject,
  project
}) {

  console.debug('PrjCardVert');
  const { 
    id,
    name,
    image,
    description, 
    feedbackRequest, 
    createdAt, 
    likesCount,
    prjCommentsCount,
    creator,
    tags
  } = project;

  const dateCreated = pgTimeToDate(createdAt);

  const classes = useStyles();

  return (
    <Card 
      elevation={3}
    >
      <MuiLink href={`/projects/${id}`}>
        <CardMedia 
          component="img" 
          image={image} 
          alt=""
        />
      </MuiLink>
      <Divider variant="fullWidth" />
      <CardContent>
        <Box mb={2}>
          <Typography
            component="h3"
            variant="subtitle2"
            className={classes.date}
          >
            Created: {dateCreated}
          </Typography>
        </Box>

        <Typography 
          component="h3"
          variant="subtitle1"
        >
          <MuiLink href={`/projects/${id}`}>
            <Box 
              fontSize="h6.fontSize"
              fontWeight="fontWeightBold"
              lineHeight="normal"
              paddingBottom="1rem"
            >
              {name}
            </Box>
          </MuiLink>

        </Typography>

        <Typography variant="body1">
          {description}
        </Typography>
      </CardContent>

      <CardActions 
        className={classes.actions} 
      >
        <Tags tags={tags} />

        <Box display='flex' align-items='center'>
          <IconButton 
            aria-label="like" 
            className={classes.iconButton} 
            onClick={toggleLikeProject}
          >
            <FavoriteBorderIcon/>
          </IconButton>
          <Box mx={1}>{likesCount}</Box>
          <ModeCommentOutlinedIcon mx={8}/>
          <Box ml={1}>{prjCommentsCount}</Box>
        </Box>
      </CardActions>

      <Divider variant="middle"/>

      <CardContent 
        display='flex' 
        justify-content="space-between" 
        align-items="center"
      >
        <Box display='flex' justify-content="space-between">
          <Avatar 
            src={creator.photoUrl} 
            className={clsx(classes.large, classes.avatar)} 
            alt="project creator" 
          />
          <Box 
            display="flex"
            alignItems="center"
            fontWeight="fontWeightBold"
            className={classes.creatorName}
          >
            <Typography>
              <ReactRouterDomLink 
                className={classes.creatorName} 
                to={`/users/${creator.id}`}
              >
                {creator.firstName} {creator.lastName}
              </ReactRouterDomLink>
            </Typography>
          </Box>
        </Box>
      </CardContent>

      <Divider variant="middle" />
      <CardContent>
        <Typography>
         {feedbackRequest}
        </Typography>
      </CardContent>
    </Card>
  );
}

export default PrjCardVert;