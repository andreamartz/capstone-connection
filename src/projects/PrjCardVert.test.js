import React from 'react';
import { render } from '@testing-library/react';
import PrjCardVert from './PrjCardVert';
import { MemoryRouter } from 'react-router';
import { UserProvider } from '../testUtils';

const project = {
	id: 3,
	name: 'Water Mate',
	image:
		'https://res.cloudinary.com/wahmof2/image/upload/v1628103099/capstone_connections/projects_capstone_connections/water-mate-Aimee-Wildstone.png',
	repoUrl: 'https://github.com/awildstone/Water-Mate',
	siteUrl: 'https://water-mate.herokuapp.com/',
	description:
		"Water Mate is a tool to help you organize your plant collection, keep track of the health of your plants, and help remind you when it's time to water your plants. You have full control over how your collection is organized, so organize your collection in a way that makes sense for you!",
	feedbackRequest: null,
	createdAt: '2021-08-05T13:19:04.891Z',
	lastModified: '2021-08-05T13:19:04.891Z',
	creator: {
		id: 3,
		username: 'awildstone',
		firstName: 'Aimee',
		lastName: 'Wildstone',
		photoUrl:
			'https://res.cloudinary.com/wahmof2/image/upload/v1628101940/capstone_connections/users_capstone_connections/Aimee-Wildstone.jpg',
	},
	likesCount: 0,
	currentUsersLikeId: null,
	tags: [{ id: 4, text: 'API' }],
};

test('it renders without crashing', function () {
	render(
		<MemoryRouter>
			<UserProvider>
				<PrjCardVert project={project} />
			</UserProvider>
		</MemoryRouter>,
	);
});

test('it matches snapshot', function () {
	const { asFragment } = render(
		<MemoryRouter>
			<UserProvider>
				<PrjCardVert project={project} />
			</UserProvider>
		</MemoryRouter>,
	);
	expect(asFragment()).toMatchSnapshot();
});
