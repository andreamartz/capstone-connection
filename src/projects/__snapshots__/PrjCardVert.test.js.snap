// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root MuiPaper-elevation3 MuiPaper-rounded"
  >
    <a
      href="/projects/3"
    >
      <img
        alt=""
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
        src="https://res.cloudinary.com/wahmof2/image/upload/v1628103099/capstone_connections/projects_capstone_connections/water-mate-Aimee-Wildstone.png"
      />
    </a>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiCardContent-root"
    >
      <div
        class="MuiBox-root MuiBox-root-26"
      >
        <h3
          class="MuiTypography-root makeStyles-date-22 MuiTypography-subtitle2"
        >
          Created: August 5, 2021
        </h3>
      </div>
      <div
        class="MuiBox-root MuiBox-root-27"
      >
        <div
          class="MuiBox-root MuiBox-root-28"
        >
          <a
            class="makeStyles-link-25"
            href="/projects/3"
          >
            Water Mate
          </a>
        </div>
        <div
          class="MuiBox-root MuiBox-root-29"
        />
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Water Mate is a tool to help you organize your plant collection, keep track of the health of your plants, and help remind you when it's time to water your plants. You have full control over how your collection is organized, so organize your collection in a way that makes sense for you!
      </p>
    </div>
    <div
      class="MuiCardActions-root makeStyles-actions-19 MuiCardActions-spacing"
    >
      <div
        class="MuiBox-root MuiBox-root-30"
      >
        <div
          class="MuiBox-root MuiBox-root-31"
        >
          <span
            class="tag"
          >
            <p
              class="MuiTypography-root API MuiTypography-body1"
            >
              API
            </p>
          </span>
        </div>
      </div>
      <div
        align-items="center"
        class="MuiBox-root MuiBox-root-32"
      >
        <button
          aria-label="like"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-21"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiBox-root MuiBox-root-33"
        >
          0
        </div>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          mx="8"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 17.17L18.83 16H4V4h16v13.17zM20 2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h14l4 4V4c0-1.1-.9-2-2-2z"
          />
        </svg>
        <div
          class="MuiBox-root MuiBox-root-34"
        />
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-middle"
    />
    <div
      align-items="center"
      class="MuiCardContent-root"
      display="flex"
      justify-content="space-between"
    >
      <div
        class="MuiBox-root MuiBox-root-35"
        justify-content="space-between"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular makeStyles-large-24 makeStyles-avatar-23"
        >
          <img
            alt="project creator"
            class="MuiAvatar-img"
            src="https://res.cloudinary.com/wahmof2/image/upload/v1628101940/capstone_connections/users_capstone_connections/Aimee-Wildstone.jpg"
          />
        </div>
        <div
          class="MuiBox-root MuiBox-root-36"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            <a
              class="makeStyles-link-25"
              href="/users/3"
            >
              Aimee Wildstone
            </a>
          </p>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-middle"
    />
    <div
      class="MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      />
    </div>
  </div>
</DocumentFragment>
`;
